File: src/main/java/com/kmarzecki/communicator/repository/RepositoryConfiguration.java
package com.kmarzecki.communicator.repository;

import lombok.AllArgsConstructor;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.jdbc.datasource.DataSourceTransactionManager;
import org.springframework.transaction.annotation.EnableTransactionManagement;

import javax.sql.DataSource;

/**
 * Repository configuration
 */
@Configuration
@AllArgsConstructor
@EnableTransactionManagement
public class RepositoryConfiguration {
    private final DataSource dataSource;

    /**
     * @return TransactionManager bean
     */
    @Bean
    public DataSourceTransactionManager transactionManager() {
        return new DataSourceTransactionManager(dataSource);
    }
}

File: src/main/kotlin/com/kmarzecki/trollo/repository/RepositoryConfiguration.kt
package com.kmarzecki.trollo.repository

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.jdbc.datasource.DataSourceTransactionManager
import org.springframework.transaction.annotation.EnableTransactionManagement
import javax.sql.DataSource

/**
 * Repository configuration
 */
@Configuration
@EnableTransactionManagement
class RepositoryConfiguration(
        private val dataSource: DataSource
) {

    /**
     * @return TransactionManager bean
     */
    @Bean
    fun transactionManager(): DataSourceTransactionManager {
        return DataSourceTransactionManager(dataSource)
    }
}